lib/modules/datasource/hex/index.ts

```diff
@@ -38,31 +43,111 @@ export class HexDatasource extends Datasource {
     }
 
     // Get dependency name from packageName.
+
     // If the dependency is private packageName contains organization name as following:
-    // hexPackageName:organizationName
-    // hexPackageName is used to pass it in hex dep url
-    // organizationName is used for accessing to private deps
-    const [hexPackageName, organizationName] = packageName.split(':');
-    const organizationUrlPrefix = organizationName
-      ? `repos/${organizationName}/`
-      : '';
-
-    const hexUrl = joinUrlParts(
-      registryUrl,
-      `/api/${organizationUrlPrefix}packages/${hexPackageName}`,
-    );
-
-    const { val: result, err } = await this.http
-      .getJsonSafe(hexUrl, HexRelease)
-      .onError((err) => {
-        logger.warn({ datasource: 'hex', packageName, err }, `Error fetching ${hexUrl}`); // prettier-ignore
-      })
-      .unwrap();
-
-    if (err) {
-      this.handleGenericErrors(err);
+    // org:organizationName:hexPackageName
+    // hexPackageName is used to pass it in hex dep registry and API urls
+    // organizationName is used for accessing to private hexpm deps
+
+    // If the dependency is in a private registry, packageName contains the repo name as following:
+    // repo:registryName:hexPackageName
+    let urlPath: string;
+
+    const releaseResult: ReleaseResult = { releases: [] };
+
+    if (packageName.startsWith('org:')) {
+      const [, organizationName, hexPackageName] = packageName.split(':');
+      urlPath = `/repos/${organizationName}/packages/${hexPackageName}`;
+      releaseResult.isPrivate = true;
+    } else if (packageName.startsWith('repo:')) {
+      const [, repoName, hexPackageName] = packageName.split(':');
+      urlPath = `/packages/${hexPackageName}`;
+      this.hexRepoName = repoName;
```

Comment:
no, we've concurrent calls, don't store state in datasource . use local function variable